apertium_apy/handlers/translate.py

```diff
@@ -157,26 +157,38 @@ def translate_and_respond(self, pair, pipeline, to_translate, mark_unknown, nosp
         before = self.log_before_translation()
         translated = yield pipeline.translate(to_translate, nosplit, deformat, reformat)
         self.log_after_translation(before, len(to_translate))
-        self.send_response({
+        self.clean_pairs()
```

Comment:
This means that we will clean pairs for each individual q. Is that what we want?

Reply:
Honestly, I don't clearly get what clean pairs mean. Can you please explain this function.

Reply:
It restarts pipes that are highly contentious and old. Take a look at the code path and it should be relatively clear. Honestly, I think `translate_and_respond` can be removed and its functionality collapsed into `get` so that cleaning pairs and logging can be done once instead of many times.

Reply:
If that's the case, then I think clean pairs can be done in `get` only. Also, since `gen.multi_future` guarantees resolving futures in parallel, isn't that more efficient than collapsing `translate_and_respond` into `get`?

Reply:
They're not mutually exclusive. Don't use the `yield` from `translate_and_respond` and you get a future.

Reply:
So what do you suggest we do about this? and should I put the clean pairs in `get` only?

Reply:
Collapse as suggested and yes.